apps/service/src/rgbpp/types.ts

```diff
@@ -41,3 +43,41 @@ export interface RgbppCkbTxHashReq {
 export interface BtcTxSendReq {
   txHex: Hex;
 }
+
+export interface RgbppTransferAllReq {
+  ckb: {
+    // The transferred RGB++ xUDT type script args
+    xudtTypeArgs: Hex;
+    // The CKB transaction fee rate in hex format, default value is 0x44c (1100)
+    feeRate?: Hex;
+  };
+  btc: {
+    // The list of BTC addresses to provide RGB++ xUDT assets
+    // All available amounts of the target asset (specified by ckb.xudtTypeArgs) will be included in the transfers
+    // However, if more than 40 cells are bound to the same UTXO, the amounts within those 40 cells are excluded
+    assetAddresses: string[];
+    // The BTC address for paying all the transaction costs, but not provide any RGB++ assets
+    fromAddress: string;
+    // The BTC address for receiving all the RGB++ assets
+    toAddress: string;
+    // The public key of sender BTC address, must fill if the fromAddress is a P2TR address
+    fromPubkey?: string;
+    // The map helps find the corresponding public key of a BTC address,
+    // note that you must specify a pubkey for each P2TR address in assetAddresses/fromAddress
+    pubkeyMap?: AddressToPubkeyMap;
+    // The BTC address to return change satoshi, default value is fromAddress
+    changeAddress?: string;
+    // The fee rate of the BTC transactions, will use the fastest fee rate if not specified
+    feeRate?: number;
+  };
+}
+
+export interface RgbppTransferAllRes extends Omit<RgbppTransferAllTxsResult, 'transactions'> {
```

Comment:
> Why does a key with 0x-hex-string appear?

@duanyytop, the hex keys are in the `RgbppTransferAllTxsResult` type for the summary info of the XUDT assets: https://github.com/ckb-cell/rgbpp-sdk/blob/develop/packages/rgbpp/src/rgbpp/summary/asset-summarizer.ts#L20

Reply:
I see. LGTM